drakcore/drakcore/frontend/src/AnalysisLogs.js

```diff
@@ -0,0 +1,352 @@
+            height={height}
+            width={width}
+            itemCount={lines}
+            itemSize={30}
+            onItemsRendered={onItemsRendered}
+            ref={ref}
+          >
+            {(props) => {
+              const lineIndex = props.index;
+
+              const chunkIndex = findChunkIndex(lineIndex);
+              const chunk = chunks[chunkIndex];
+              if (chunk === undefined) {
+                return children({
+                  entry: undefined,
+                  index: lineIndex,
+                  style: props.style,
+                });
+              }
+              const lineOffset = index ? index.markers[chunkIndex].line : 0;
+              const entry = chunk[lineIndex - lineOffset];
+              return children({
+                entry,
+                index: lineIndex,
+                style: props.style,
+              });
+            }}
+          </List>
+        )}
+      </InfiniteLoader>
+    );
+  };
+  return <AutoSizer>{loader}</AutoSizer>;
+}
+
+const SERVICE_LOGS = ["drakrun.log", "drak-postprocess.log"];
+
+const DrakvufRow = ({ entry, style }) => {
+  let content;
+  let displayed = { ...entry };
+  if (entry === undefined) {
+    content = <>Loading...</>;
+  } else {
+    content = (
+      <code style={{ whiteSpace: "nowrap" }}>{JSON.stringify(displayed)}</code>
+    );
+  }
+
+  return (
+    <div style={style} className="d-flex flex-row align-content-stretch">
+      <div className="p-1">{content}</div>
+    </div>
+  );
+};
+
+function formatTimestamp(timestamp) {
+  const newDate = new Date();
+  newDate.setTime(timestamp * 1000);
+  const dateString = newDate.toUTCString();
+  return dateString;
+}
+
+const ServiceRow = ({ entry, index, style }) => {
+  if (entry === undefined) {
+    return <>Loading...</>;
+  }
+  return (
+    <div style={style} className="d-flex flex-row align-content-stretch">
+      <div className="p-1">
+        <div>
+          <code style={{ whiteSpace: "nowrap" }}>{entry.message}</code>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+class AnalysisLogs extends React.Component {
+  constructor(props) {
+    super(props);
+    this.analysisID = this.props.match.params.analysis;
+
+    this.state = {
+      currentLog: null,
+      currentTab: "drakvuf",
+      logList: [],
+      index: null,
+    };
+
+    this.selectionChanged = this.selectionChanged.bind(this);
+    this.getLogChunk = this.getLogChunk.bind(this);
+    this.setTab = this.setTab.bind(this);
+  }
+
+  setTab(tab) {
+    this.setState({ currentTab: tab });
+  }
+
+  async getLogChunk(from, to) {
+    const chunks = await api.getLogRange(
+      this.analysisID,
+      this.state.currentLog.split(".")[0],
+      from,
+      to
+    );
+    if (chunks) {
+      return chunks.request.response.split("\n").map((line) => {
+        try {
+          return JSON.parse(line);
+        } catch (e) {
+          console.log(e);
+        }
+        return undefined;
```

Comment:
We don't do that here. Never use `undefined` as a value. Use `null` if you have. Otherwise you'll spend +inf time debugging stuff.